Documentation/ExtensionArchitecture/ComposerJson/Index.rst

```diff
@@ -188,10 +188,32 @@ Define namespace - path mapping for PSR-4 autoloading.
 In TYPO3 we follow the convention that all classes (except test classes)
 are in the directory :file:`Classes`.
 
+extra
+-----
+
+(*required*)
+
+The extra `typo3/cms` section is used to provide a TYPO3 extension_key for the package.
+If not provided, the package-key will be used with all dashes (`-`)
+replaced by underscores (`_`) to follow TYPO3 and Packagist conventions.
+
+Not providing this property will emit a deprecation notice and will fail in future versions.
+
+So, the following section can be provided, but the default will result in
```

Comment:
> So, the following section can be provided, but the default will result in the same thing

What do you think about removing this sentence? If it is always required and will emit deprecation notice if not provided, I think it does not really matter what the default is.

Reply:
Good point. I agree